csi/controller.go

```diff
@@ -1107,6 +1133,41 @@ func (s *OsdCsiServer) DeleteSnapshot(
 	return &csi.DeleteSnapshotResponse{}, nil
 }
 
+func (s *OsdCsiServer) deleteCloudBackup(
+	ctx context.Context,
+	req *csi.DeleteSnapshotRequest,
+) (*csi.DeleteSnapshotResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetSecrets())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
+	}
+
+	credentialID := locator.VolumeLabels[osdSnapshotCredentialIDKey]
+
+	backupID := req.GetSnapshotId()
+
+	// Delete snapshot
+	resp, err := cloudBackupClient.Delete(ctx, &api.SdkCloudBackupDeleteRequest{
+		BackupId:     backupID,
+		CredentialId: credentialID,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Aborted, "Failed to delete cloud snapshot: %v", err)
+	}
+
+	if resp != nil {
+		return &csi.DeleteSnapshotResponse{}, nil
+	}
+
+	return nil, status.Errorf(codes.Aborted, "unexpected error occured: %v", err)
```

Comment:
This can just be simplified to `return nil`.

If we didn't get an error on line 1160, we know deleted succeeded and `err` will be nil.